pkg/ast/spl/spl.peg

```diff
@@ -2157,6 +2227,507 @@ AggSum <- "sum" L_PAREN "eval" L_PAREN boolComparisonExpr:BoolComparisonExpr R_P
     return agg, nil
 }
 
+// Todo: implement the processing logic
+AggEstdc <- "estdc" L_PAREN "eval" L_PAREN boolComparisonExpr:BoolComparisonExpr R_PAREN R_PAREN {
+    valueExpr := &structs.ValueExpr {
+        ValueExprMode: structs.VEMBooleanExpr,
+        BooleanExpr: boolComparisonExpr.(*structs.BoolExpr),
+    }
+
+    agg := &structs.MeasureAggregator {
+        MeasureCol: "",
+        MeasureFunc: utils.Estdc,
+        StrEnc: string(c.text),
+        ValueColRequest: valueExpr,
+    }
+
+    return agg, nil
+}
+/ "estdc" L_PAREN field:FieldName R_PAREN {
+    agg := &structs.MeasureAggregator {
+        MeasureCol: field.(string),
+        MeasureFunc: utils.Estdc,
+    }
+
+    return agg, nil
+}
+
+// Todo: implement the processing logic
+AggEstdcError <- "estdc_error" L_PAREN "eval" L_PAREN boolComparisonExpr:BoolComparisonExpr R_PAREN R_PAREN {
+    valueExpr := &structs.ValueExpr {
+        ValueExprMode: structs.VEMBooleanExpr,
+        BooleanExpr: boolComparisonExpr.(*structs.BoolExpr),
+    }
+
+    agg := &structs.MeasureAggregator {
+        MeasureCol: "",
+        MeasureFunc: utils.EstdcError,
+        StrEnc: string(c.text),
+        ValueColRequest: valueExpr,
+    }
+
+    return agg, nil
+}
+/ "estdc_error" L_PAREN field:FieldName R_PAREN {
+    agg := &structs.MeasureAggregator {
+        MeasureCol: field.(string),
+        MeasureFunc: utils.EstdcError,
+    }
+
+    return agg, nil
+}
+
+// Todo: implement the processing logic
+AggExactPerc <- "exactperc" numStr:(FloatAsString / IntegerAsString) L_PAREN "eval" L_PAREN boolComparisonExpr:BoolComparisonExpr R_PAREN R_PAREN {
+    valueExpr := &structs.ValueExpr {
+        ValueExprMode: structs.VEMBooleanExpr,
+        BooleanExpr: boolComparisonExpr.(*structs.BoolExpr),
+    }
+
+    agg := &structs.MeasureAggregator {
+        MeasureCol: "",
+        MeasureFunc: utils.ExactPerc,
+        StrEnc: string(c.text),
+        ValueColRequest: valueExpr,
+        Param: numStr.(string),
+    }
+
+    return agg, nil
+}
+/ "exactperc" numStr:(FloatAsString / IntegerAsString) L_PAREN field:FieldName R_PAREN {
+    agg := &structs.MeasureAggregator {
+        MeasureCol: field.(string),
+        MeasureFunc: utils.ExactPerc,
+        Param: numStr.(string),
+    }
+
+    return agg, nil
+}
+
+// Todo: implement the processing logic
```

Comment:
For these `Todo`s, you mean implement the logic in the backend to actually support these commands, right? If so, I don't think you need these TODOs in this file, since the thing to do wouldn't be implemented in this file. A better place would be in segaggs.go where we do the other processing, we can check the `MeasureFunc` and add some sort of `Not implemented` message

Reply:
Yes, but I'm actually not sure where exactly to implement these methods because they could be in `segaggs.go, segresults.go, evalaggs.go`, or possibly other files. What should I do in this situation?

Reply:
Hmm good point. I think it should flow through segaggs.go anyways though, even if it's not implemented there. So could you add a `Not implemented` message there and later when we actually implement these we would remove that message but can implement it wherever it actually should be implemented? Do you think that would work?

Right now it I think these queries are silently failing because they're ignoring the new data being parsed. This is going to lead to a lot of customer confusion because running the query doesn't give any indication that we don't support those options/functions, but the results won't be correct in many cases.